pkg/db/interface.go

```diff
@@ -455,18 +455,60 @@ func (d *dbConn) GetWorkerLogs(id string, opts *GetWorkerLogOpts) ([]*WorkerLog,
 	return result, nil
 }
 
-func (d *dbConn) getWorkerLastlog(id string, value *string) (*time.Time, error) {
-	var lastTimestamp string
+func (d *dbConn) getWorkerLastlogs(id string) (time.Time, []*WorkerLog, error) {
+	var timeStr string
+	var timeVal time.Time
 	var err error
 
-	if value != nil {
-		row := d.db.QueryRow("SELECT time, value FROM worker_lastlogs WHERE worker_id = ?", id)
-		err = row.Scan(&lastTimestamp, value)
-	} else {
-		row := d.db.QueryRow("SELECT time FROM worker_lastlogs WHERE worker_id = ?", id)
-		err = row.Scan(&lastTimestamp)
+	// Use LEFT JOIN to ensure a result even if there's no matching
+	// in worker_metrics.
+	rows, err := d.db.Query(
+		"SELECT worker_lastlogs.time, name, value FROM worker_lastlogs "+
+			"LEFT JOIN worker_metrics ON (worker_lastlogs.worker_id = worker_metrics.worker_id AND worker_lastlogs.time = worker_metrics.time) "+
```

Comment:
Why not using ` to make a multiple line string?

Reply:
Fixed.